### DISCUSSION TITLE
Multiple Exposure Proposal Without Lifting the Plate

### DISCUSSION THREAD

**Discussion by joandescarga** [sn4k3/UVtools#848] [Ideas]
I propose a new feature for UVtools that significantly speeds up printing and enhances quality. 
The idea is to slice at 0.02 layers and then only lift the build plate every 5 layers, or 0.1mm. This means five layers are sequentially exposed with a decreasing exposure sequence, for example, 0.5, 0.5, 0.4, 0.4, 0.4, totaling 2.2 for the 0.1mm layer. After every 5 layers, the plate lifts, and another 5 are exposed without lifting. This greatly reduces exposure time while achieving 0.02 resolution in a 0.1 layer. I've tested this by manually modifying it with UVtools on a Halot Mage, and it works very well. The only thing to note is that all layers must have a minimum lift value.

**Comment by sn4k3** [comment]
This can be easy done with a script. The problem of this method is that few printers will be able to stay in same position. For most there is a mandatory lift. GCode printers can stay in place without lift. Lift per layer is still not possible on some printers.
But if you not lifting per 5 layers or have a very low value, how the model detach and get a fresh coat of resin?

**Reply by joandescarga** [reply]
Thank you, Tiago, for the response. It's the same as in the normal process only that the exposure changes layers each time, the plate lifts by 0.1, but instead of exposing a new layer for 3 seconds, it makes five sequential exposures of 5 different layers during those 3 seconds without lifting the plate. Can you guide me on how I can create that script? It would help me a lot in my tests

**Reply by Hazoooka** [reply]
Try multi exposure, perhaps this will do the trick. 
Prepare the print for 100um layers and set up 5 exposures for the time of your choosing. 
You can apply this to all layars.

**Comment by Hazoooka** [comment]
Don’t be sorry, that’s the learning curve for all of us 🙂

**Reply by joandescarga** [reply]
Thanks Hazoooka, but finally I think that I was not wrong. Lets say you have a model 10 mm tall and if you slice at 0.1 you have 100 layers at lets say 3s per layer you have 100 lifts and the precision is 0.1. Same model sliced at 0.02 you have 500 layers and exposing at 0.6s every 0.02 layer but only lifting every 5 layers you have same 100 liftings and same exposition time but the precision of 0.02 insted 0.1. 
I tink that I was explainig bad before but the system works.
 I have to find test models to see the diference in smoth surfaces

**Reply by Hazoooka** [reply]
If you set the layer for 100um you will have precision of 100um, not 20um, regardless of how many times you expose it.
Moreover, it’s possible that all the exposure runs add up and you will get even less quality.
If you are looking for the best quality there is no workaround other than finding the highest layer thickness fitting your needs.

For the smooth surface try the full greyscale range, AA 2 and blur 4.
At least thats ho it works for me on Phrozen printer.

**Reply by joandescarga** [reply]
Please look what I'm saying I thing is correct. 10mm 0.1mm 100 layers - 0.02mm 500 layers quality obviously better . The difference is that in both cases I'm lifting 100 times but instead of exposing 1 layer at a time 3s I'm exposing five different layers 0.6s without lifting

![slice](https://github.com/sn4k3/UVtools/assets/129888231/205456f4-a584-4d4d-a911-3822672cd18c)

**Comment by Hazoooka** [comment]
If you are exposing without lifting you are still curing one layer, no matter how many times you run the exposure.
That's the main problem with this theory.
You will not get the slopy effect because you are curing the edges over and over again with every exposure pass, so you are curing the slopes and they become part of the main shape we might say. The edges will be straight.
The longer you cure the edges the less slope effect we get.

Exposure finder is for finding the perfect balance between the quality and eficiecy of the print.
For Phrozen 8k resin for example the perfect balance for 20um layer is 1.4s, at lease that's what I have found. If you go for 1.3s the model comes out a bit too flexible.

[edit]
1.4s for the best details that is, for the smooth surface it's more.

**Comment by Hazoooka** [comment]
... however if you edit your slices properly, or find good settings for multi exposure to avoid exposing the edges more than once, then perhaps it may work.
This I do not know, I haven't done this stunt yet.

**Comment by joandescarga** [comment]
Creo que no me se explicar... no estoy exponiendo los mismos bordes cada vez porque cada capa es distinta. Si hiciera exposiciones multiples de la misma capa no tendria sentido pero cada capa de 0.02 tiene bordes diferentes de la anterior por lo tanto no estoy curando los mismos bordes cada vez. Una capa de 0.1 tiene un borde vertical con respecto a la siguiente por ejemplo en una inclinacion de 45 grados pero 5 capas de 0.02 represetan cinco bordes diferentes para un mismo desplazamiento de 0.1mm

**Comment by joandescarga** [comment]
I think I'm not explaining myself well... I'm not exposing the same edges each time because each layer is different. If I were to make multiple exposures of the same layer, it wouldn't make sense, but each 0.02 layer has different edges from the previous one, so I'm not curing the same edges every time. A 0.1 layer has a vertical edge with respect to the next one, for example, at a 45-degree angle, but 5 layers of 0.02 represent five different edges for the same 0.1mm displacement

**Comment by Hazoooka** [comment]
> I'm not exposing the same edges each time because each layer is different

That clears the picture :)
Contracting each next layer by the slope resolution, say 1px, is not a case because it will work within pixels, and you want to achieve the slope within exactly one pixel sometimes.
If I understand correctly you will need to dim every next layer to possibly produce the desired slope, make the grayscale yourself in other words.
This is exactly what the blur is for, it's dimming the edges so we do not have to do it ourselves by hand.

But perhaps I am not fully following the plan again.

[edit]
typos corrected

**Comment by Hazoooka** [comment]
I have actually tried this once, making my own slices few weeks after receiving the printer, when I have realised that 10um layer is not enough for the smooth top oval. Also discussed it with Chitubox.
I have made my own oval part by rendering every next 1um layer myself and then combining them in PS into one 10um layer slice.
I was working within one layer, exactly what I think you are proposing, but instead of multi exposure in addition to custom edited slices I have combined the slices into one.
It didn't work. At the and of the day I always got steps, even using small exposure times.

I've dropped this idea as not efficient in favour of more promising things.
Perhaps it is possible, but would require so much tinkering, beyond reason, finding the proper grey color for every next sub-slice, then over inflating the stack by 'n' to fit extra layers. And then, on top of all this you will need to apply every single sub-slice sepately which would take ages to complete.
However if one could find the right greyscale for each sub-slice this might work? I didn't find it, so who knows, maybe I failed because I didn't do it properly.
This was my first contact with this kind of resin printer. Before I worked on Stratasys, all in one resin printer, not requiring any rocket science knowledge.

[edit]
typos corrected

**Comment by sn4k3** [comment]
IMO The best approach is to use the dynamic layer height if your printer supports. Slice with 0.01mm and let UVtools optimize stack of layers up to 0.10mm.

If you want to look into scripting see here: https://github.com/sn4k3/UVtools/tree/master/Scripts/UVtools.ScriptSample

**Comment by Hazoooka** [comment]
But scaling layers from 20um to 100um will not fix the problem, they will be still 100um, poor quality print.
The question is how to fake 20um layers using 100um ones and keep the quality of 20um.
I don't think it's doable.

**Comment by joandescarga** [comment]
I think I still don't explain myself well.

If I have a 2mm model with 100 layers of 20um, I have 100 steps of 20um, which will be the same as 20 layers of 100um, but in this case I have 20 steps of 100um, that is, of worse quality.

Now if instead of lifting the plate every 20um and exposing each layer, I lift the plate 100um, that is, the space for 5 layers of 20um, and without moving it, I sequentially expose the five layers of 20um that are different, I will have a layer of 100um made up of five layers. of 20um but I have only lifted the plate once.

Sorry, I don't know how to explain better.

**Comment by Hazoooka** [comment]
Show us the sub-slices.

**Comment by joandescarga** [comment]
ok, tomorrow if I can

**Comment by Hazoooka** [comment]
I understand you, you have 100um layer, but made from 20um slices.
Still, you have 100um layer, you will not get quality of 20um one this way.
You will have a quality of 100um.

And that's exactly what I have done.
I have divided my 10um layer by 10 and made 1um sub-slices.

Now that I think of it, given my present experience I think I will give it one more go.
The problem of the visible steps appears only at some angle, quite sharp one, small, can't remember, a few degrees, up to around 10 deg. Above that point blur does the job quite well for the 10um layers.
Anyway, top oval parts give too much gap between the edges of the subsequent slices that needs to be blurred correctly.
Slicers apply default blur, but it needs to be streched smoothly, a gradient from one edge to the other.
The small gaps at the steep angles are pretty well handled, the small, sharp angles produce steps.
I think the problem is the partly cured resin at the edges that actually simulates slopes. With too much gap it's not sticking enough.

[Edit]
By too much gap I mean area of the partly cured resin larger than the standard 1px or 2px, that may peal off easily. That's why I failed for the first time.

But it's nice to test.

**Comment by joandescarga** [comment]
Thank you Hazooka, this is what I find hard to understand: "I understand you, you have 100um layers, but made from 20um slices. Still, you have a 100um layer, you will not get the quality of a 20um one this way. You will have the quality of a 100um."

If I have 20um layers, the edge of each layer is only exposed once, and therefore I should have the five 20um steps. However, in a 100um layer, it's as if I only exposed the fifth layer each time, and therefore I would have 100um steps. Am I misunderstanding this?

**Comment by Hazoooka** [comment]
When you are exposing a layer, regardless of the thickness, you are exposing whole, all the way through.
If you apply 1/5 exposure it will be cure at 20%. 
When you apply 1/5 exp five times the layer will be properly cured. 

In your case, the edges, you just need to sort out what to cure first.

**Comment by Hazoooka** [comment]
Could you share your slices?

**Comment by Hazoooka** [comment]
As for the quality, you want to achieve 20um fine surface with 100um layers. 
in other words you want to make low steps with a high layer. 
For this you need edges to be stepped, I call it sloped. 
For edges to be stepped you are dividing a high step by 5. 
Every subsequent step exp pass will cure the center of your layer by 20% only, resulting in a wet, under-cured edges. 
If the area of the edge below is too large the under-cured, partly wet resin will just not stick.
So in the end you may end up with a sharp edge of the 100um layer instead of five 20um steps with all the under-cured resin enjoying a bath in the vat. 
And then instead of having 20um qualify you get 100um quality. 

By quality I mean the visible layers, steps that show up on the surface depending on the layer hight.
The smaller the hight the finer the surface. 

In your case you may get five times worse quality

PS
Writing on the iPhone 5 traveling is a bit annoying sometimes. 
It seem to be taking as much time as Treabeards talking to each other :)

**Comment by joandescarga** [comment]
Thank you, Hazooka, I finally understood what the problem is. When I expose the first layer, I am exposing the entire 100µm, not just the first 20µm, therefore the subsequent layers will harden the layer but the profile of the 100µm layer is determined by the first one which will have some underexposed pixels. The only case in which it would work is on concave surfaces where the layers are retracting, in this case, I do believe that the 20µm steps would come out with the last one underexposed.

I appreciate your kindness and patience!

**Comment by Hazoooka** [comment]
My pleasure.

Note that the blur is doing all you want to achieve automatically by default, however up to some point.
The steeper the angle the better surface finish, also to some point .
For 10um on my printer I have found that 10deg is already enough to smooth the horisontal steps.
This unfortunately leads to positioning object tilted by 10 deg which sometimes can lead to 20h print and more, like for  example in case of a 1:700 destroyer hull.

But the subject is still interesting.
Perhaps it's possible to gain some degrees by editing the slices and get it to work in 5 deg for example.
I would be interested to test it.

**Comment by joandescarga** [comment]
I'm thinking of trying something else that seems challenging, but I like to push the limits... Is there a way to change the layer order, or the exposure sequence? By doing this, I believe I could achieve the quality effect I'm looking for with 20 layers at 100 microns. Do you think it's feasible? Is there any way to test it with UVTools?
![20240309_194327](https://github.com/sn4k3/UVtools/assets/129888231/e0d6932d-c3f2-4a74-bf9c-c5abe60a8ef1)

**Reply by Hazoooka** [reply]
In my opinion testing is the only way, there is nothing else.
I have learned what I know about resin print exactly doing so, pushing the limits :)

The layer order is restricted by the shape you want to get.
But manually editing the slices is limited by pixels.
If you have a fairly flat slope this may work, may be enough room for the gradient.
[edit] but again, partly cured resin :(
But if the slope is steep, i think it's better to rely on the slicer's tools.

Chitubox for example has the best aa in my opinion, the surfaces are really smooth, while Lychee is always producing jagged aa edges. You can slice the same model and see.

**Reply by Hazoooka** [reply]
Missed my answer to your drawing.
Yes, the second way seems better, actually it may work.
No other way but to test :).

How do you plan to attach all the sub-slices to subsequent layers? Because by hand it will take a lot of time. Perhaps someone could write the script automatically attaching the sub-slices to proper layers, it would be a matter of naming the slices perhaps to be recognised by the script.

**Comment by Hazoooka** [comment]
> The only case in which it would work is on concave surfaces where the layers are retracting, in this case, I do believe that the 20µm steps would come out with the last one underexposed.



> The only case in which it would work is on concave surfaces where the layers are retracting, in this case, I do believe that the 20µm steps would come out with the last one underexposed.

The best is to use the default tools like greyscale, aa, blur and UV tools, and only if those do not work turn to manual editing

**Comment by aWildTomAppeared** [comment]
I had this last year, I called it 'semi-continuous printing'  I tested it on my saturn 1 but it was just to iffy to go through with here are the issues:

1. like sndk3 said, many printers have a minimum lift, on the saturn 1 that is 0.01mm which is basically nothing, but I can also disable that in the g-code (M8070 to Z0)

2. curing resin isn't linear, and it's even less so with this because you are wanting to cure at different distances and above layers that have just been cured, so getting it to cure each layer the exact right takes a LOT of calibration, I did so many tests by manually changing lifts and layer heights in UVT, I made a script to make it a bit easier but it wasn't very good.  In the end I left it because my tests with 3D AA were FAR more succesful. 

3. even VERY knowledgeable who have been in this industry as long as it has been about don't know this, but I have confirmed in tests that resin when MSLA printing DOES NOT cure fep up, it is because of this that this or 3dAA is at all possible. HOWEVER, it doesn't necessarily just cure from the previous layer down either. It starts to cure from a distance above the fep and then up and down. This point certainly varies based on the z bloom of the resin, so clear resins and white resins with lots of z bloom can start curing deeper than high pigment low z bloom resins. I also think light intensity and collimation could help cure deeper, however I have not confirmed that. The new hey gears reflex printer claims to use two different uv wave lenghts to cure at different depths so aparantly that's a thing but thats a very closed system so this type of testing won't be possible there. 
In my testing I find that this distance is generally about 50um, the layer height that most resins are designed for. Because of this, doing this at 0.1mm isn't going to happen, the resun just won't cure partial layers that far from the fep and you will get delamination, the only reason you can print full layers at that height is that there is enough exposure time for the resin to cure up to the last layer and down to the fep. This is why if you try 200um printing you find jagged edges because as the resin cures towards the fep sooner than the previous layer causing a little elephants foot on each layer. 

4. To get around the cure depth limit you could try to instead say have a 50um distance and cure a 25um partial layer, and then lift 25um and keep doing this so the layers don't touch the fep, I tested this too with a bad uvt script, but it's hard to do because of curing inconsistiencies, like areas with more bloom will cure more. Also, even though it may not full cure to the fep A you still need to do a lift retract and wait to get a new layer of resin to flow in so this may reduce peel forces but with suction forces, not fully. B in some of my partial layer curing tests I find partially cured layers look normal until they are cleaned, suggesting that even if you fully cure a partial height layer, there could be some soft semi cured resin that still touches the fep and therefore a full lift may still be needed. These potential soft cured areas could be why I had troubles with testing this
